internal/controller/pipelinerollout/pipelinerollout_controller.go

```diff
@@ -347,6 +347,12 @@ func (r *PipelineRolloutReconciler) reconcile(
 			if err := r.client.Delete(ctx, pipelineRollout, &client.DeleteOptions{PropagationPolicy: &foreground}); err != nil {
```

Comment:
It looks like calling `r.client.Delete()` doesn't actually update the pipelineRollout to reflect the new Finalizer called `foregroundDeletion` and the new resource version. So, the subsequent call would sometimes fail to update. 

For some reason in the PPND test that was commented out in here, it basically would always fail the same way - in a constant cycle of never being to update. 

We can either do what I have here which is to Get the live object after doing the Update(), or we could change the Update() to a Patch().